pulpcore/plugin/stages/content_stages.py

```diff
@@ -105,17 +104,16 @@ class ContentSaver(Stage):
                                     d_content.content.q())
                             continue
                         for d_artifact in d_content.d_artifacts:
-                            artifact_already_saved = not d_artifact.artifact._state.adding
-                            if artifact_already_saved:
-                                content_artifact = ContentArtifact(
-                                    content=d_content.content,
-                                    artifact=d_artifact.artifact,
-                                    relative_path=d_artifact.relative_path
-                                )
+                            if not d_artifact.artifact._state.adding:
+                                artifact = d_artifact.artifact
                             else:
-                                raise ValueError(_(
-                                    "Attempting to save a ContactArtifact for an unsaved Artifact"
-                                ))
+                                # in the case of lazy sync, artifact could be none
```

Comment:
```suggestion
                                # set to None for lazy synced artifacts
```
I think, the comment is not yet clear enough, that lazy artifacts are actually missing at this time.